docs/en/contributing-to-appium/developers-overview.md

```diff
@@ -0,0 +1,150 @@
+* No merge commits: always rebase on top of latest master (or whatever other branch you're asking to merge into) before submitting your PR.
+* Almost all changes should have tests. Bugfixes should at the least have unit tests that prove that the bug has been fixed. New features should have unit tests and in most cases e2e tests to prove that the feature actually works. We will be happy to walk you through the test creation process. Reading the surrounding test code is a good place to start. Our CI systems usually run test coverage statistics and we will likely not merge code that decreases test coverage.
+
+If you do all of these things before submission, your code will almost
+certainly be accepted very quickly! Of course, if you're thinking of making
+a change to Appium that requires a lot of work, you might reach out to the
+developers list to make sure that the change is in line with our philosophy and
+in principle something that we'll accept before you get going.
+
+### Testing
+
+Always make sure that your changes are tested! In addition to writing unit and
+e2e tests, make sure you run existing tests before you begin to make changes
+and before you push code to be reviewed. We do have CI set up for every Appium
+repository as a safety net for reviewers to know whether the code they are
+reviewing has passed muster. Running tests in any Appium package is easy!
+Unless the README says otherwise, here are the things you can do:
+
+```
+gulp                    # watch code for transpilation and run unit tests
+gulp once               # same as above but don't watch
+gulp unit-test          # transpile and run unit tests
+gulp e2e-test           # transpile and run e2e tests
+_FORCE_LOGS=1 <command> # show module log output during test run
+```
+
+Note that we have a convention for unit test files to end in `-specs.js` and
+for e2e test files to end in `-e2e-specs.js`.
+
+### Releasing
+
+The release process for any Appium module other than the main Appium package is
+pretty straightforward (note that you will need to be an NPM owner for the
+package if you want to publish it. Ownership is managed by the Appium
+committers; talk to @jlipps or @imurchie if you believe you should be an owner
+and are not):
+
+0. `rm -rf node_modules && npm install` and run tests to make sure a clean install works.
+0. Determine whether we have a patch (bugfix), minor (feature), or major (breaking) release according to the principles of SemVer.
+0. Update the CHANGELOG and/or README with any appropriate changes and commit. Most subpackages don't have a CHANGELOG.
+0. Run `npm version <version-type>` with the appropriate version type.
+0. This will create a version tag on your branch. Push to the appropriate branch on GitHub.
```

Comment:
I would change this to:

Push the appropriate branch to github, don't forget to add the `--tags` flag to push the new version tag created by `npm version`